src/users_db.rs

```diff
@@ -126,16 +132,30 @@ fn escape(string: &str) -> String {
 impl UserBuilder {
     const MIN_PASS_LEN: usize = 8;
 
-    pub fn new() -> UserBuilder {
-        UserBuilder {
-            id: None,
-            name: String::new(),
-            email: String::new(),
-            password: String::new(),
-            secret: String::new(),
-            error: None,
-            is_admin: Some(false)
+    pub fn new(user: Option<User>) -> UserBuilder {
+        match user {
+            Some(user) =>   UserBuilder {
```

Comment:
Well, in this case we have a UserBuilder instance, not a User one. We could do a user.clone() and directly set the new fields on the cloned instance, but then we would lose the validation of these new fields that UserBuilder provides.